tools/deployment-cli-tools/ch_cli_tools/migration.py

```diff
@@ -0,0 +1,101 @@
+"""
+Utilities to perform a migration of the deployment to the latest supported version.
+"""
+import os
+import json
+
+from . import HERE
+from cloudharness_utils.constants import APPS_PATH
+from .utils import get_sub_paths, search_word_in_file, search_word_in_folder
+
+TO_CHECK = ["deploy", "tasks", "Dockerfile"]
+
+
+def perform_migration(base_root, accept_all=False):
+    all_files_detected = []
+    for file in os.listdir(os.path.join(HERE, "migrations")):
+        f = open(os.path.join(HERE, "migrations", file), "r")
+        migration_json = json.load(f)
+
+        sub_paths = []
+        if isinstance(base_root, list):
+            app_base_path = [
+                os.path.join(base_root_path, APPS_PATH)
+                for base_root_path in base_root
+            ]
+            for app_path in app_base_path:
+                sub_paths.extend(get_sub_paths(app_path))
+        else:
+            app_base_path = os.path.join(base_root, APPS_PATH)
+            sub_paths = get_sub_paths(app_base_path)
+
+        for app_path in sub_paths:
+            # Iterate the folders and files to check if they need to be migrated
+            for sub_path in TO_CHECK:
+                to_check = os.path.join(app_path, sub_path)
+                if os.path.isdir(to_check):
+                    for migration_obj in migration_json["deprecated"]:
+                        files = search_word_in_folder(to_check, migration_obj["keyword"])
+                        for file in files:
+                            file_path = os.path.join(to_check, file)
+                            print("#########################################")
+                            print(f"Running migration on {file_path}")
+                            print("#########################################")
+                            all_files_detected.append(file_path)
+                            for word_to_replace in migration_obj["to_be_replaced"]:
```

Comment:
Suggestion: 6 nested for loops are a lot and make the code difficult to read. Might extract some function to make the code more readable (inner function would be OK too)